tools/clang_tidy/lib/clang_tidy.dart

```diff
@@ -171,23 +189,86 @@ class ClangTidy {
     return repo.changedFiles;
   }
 
+  Iterable<T> _takeShard<T>(Iterable<T> values, int id, int shardCount) sync* {
```

Comment:
I think the code from here down that is reasoning about sets of commands could be made a bit more idiomatic, and maybe more readable as well, by encapsulating it in a new class (`Commands`, `CommandSet`, or `CommandShard`, or something) that can live in `lib/src/command.dart`.

Reply:
Respectfully, I don't think that will help make the code easier to read or maintain.  The small codebase already has poor oop design already and it actually made my job harder.  In objective terms adding more classes will result in more lines of code, more segmentation of the code, and more concepts you need to know to modify the code.

As an example, there is no reason for `CommandSet` to exist when `Set<Command>` (or `List<Command>`) already encapsulates the idea perfectly.  People can come into the codebase and know what that is immediately, not so with `CommandSet`.  I also can't understand from what the difference between a `CommandSet` and a `CommandShard` is.

Reply:
> As an example, there is no reason for `CommandSet` to exist when `Set<Command>` (or `List<Command>`) already encapsulates the idea perfectly.

The code in `getLintCommandsForFiles` is not operating on the `Set`, `List`, etc. in its full generality. Instead it's doing something very specific: trying to filter unwanted items out of a big set. The specific internal data structures that are used to accomplish that goal are an implementation detail. This code would be easier to read and maintain if the the implementation details were encapsulated somehow. It doesn't necessarily have to be a class. It could be more helper methods, for example. I don't have a strong opinion there. However, I think even with comments, the code in `getLintCommandsForFiles` could be improved on, and that is overall what I'm asking for.

> I also can't understand from what the difference between a `CommandSet` and a `CommandShard` is.

These were just naming suggestions for a single new class.

Reply:
Yea, I think the documentation could be better and will help with that.  I've updated the docstring in `getLintCommandsForFiles` and added one for `_takeShard`.